### PR TITLE
src/components/routes: update and refactor RouteItemDisplay and RouteItemEdit components

### PR DESCRIPTION
**Pull request by maceksimon**
Update and refactor `RouteItemDisplay` and `RouteItemEdit` components based on UI design.

__UI design__

* Update layout (elements stacked vertically instead of horizontally)
* Border, border radius and separators
* Use custom icons
* Add empty state to a RouteItemDisplay element (question-mark icon)

__Refactor__

In order to be able to use parts of the route-edit UI in calendar popup panel, refactor these elements into new components: `RouteInputTransportType` and `RouteInputDistance`
* Add tests

### CODE REVIEW

File: src/components/__tests__/RouteInputDistance.cy.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+    it('renders inputs side by side (align-bottom)', () => {
+      // input type is input distance
+      cy.dataCy('select-action')
+        .should('be.visible')
+        .find('input')
+        .should('have.value', i18n.global.t('routes.actionTraceMap'));
+      // input type and input distance are on the same line
+      cy.dataCy('section-input-action').then((element) => {
+        const offsetTop = element[0].offsetTop;
```
Comment (tmszi) [inline src/components/__tests__/RouteInputDistance.cy.js:80]: These code lines repeat (line 45-52), rewrite as func (or Cypress command), please.

File: src/components/__tests__/RouteInputDistance.cy.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+}
+
+function buttonMapTests() {
+  it('renders map button', () => {
+    cy.dataCy('input-distance').should('not.exist');
+    cy.dataCy('button-trace-map')
+      .should('be.visible')
+      .and('have.css', 'font-size', '16px')
+      .and('contain', i18n.global.t('routes.buttonTraceMap'));
+    cy.dataCy('icon-trace-map').should('be.visible');
```
Comment (tmszi) [inline src/components/__tests__/RouteInputDistance.cy.js:159]: `const iconTraceMap = 'icon-trace-map'`

File: src/components/__tests__/RouteInputDistance.cy.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    cy.dataCy('icon-trace-map').invoke('width').should('be.equal', 24);
```
Comment (tmszi) [inline src/components/__tests__/RouteInputDistance.cy.js:160]: `const iconTraceMapIconSize = 24`

File: src/components/__tests__/RouteInputTransportType.cy.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        'margin-top',
+        '0px',
+      );
+      // description not full width (is flexed)
+      cy.dataCy('description-transport')
+        .invoke('width')
+        .should('be.lessThan', 300);
+      // description and buttons are on the same line
+      cy.dataCy('description-transport').then((element) => {
+        const offsetTop = element[0].offsetTop;
```
Comment (tmszi) [inline src/components/__tests__/RouteInputTransportType.cy.js:56]: Same as my comment above.

File: src/components/__tests__/RouteInputTransportType.cy.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+      });
+    // label transport
+    cy.dataCy('label-transport')
+      .should('be.visible')
+      .and('have.css', 'font-size', '12px')
+      .and('have.css', 'font-weight', '700')
+      .and('have.color', grey10)
+      .and('contain', i18n.global.t('routes.labelTransportType'));
+    // avatar
+    cy.dataCy('avatar-transport').should('be.visible');
```
Comment (tmszi) [inline src/components/__tests__/RouteInputTransportType.cy.js:102]: `const avatarTransport = 'avatar-transport'`

File: src/components/__tests__/RouteInputTransportType.cy.js [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    cy.dataCy('avatar-transport').invoke('height').should('be.eq', 32);
+    cy.dataCy('avatar-transport').invoke('width').should('be.eq', 32);
```
Comment (tmszi) [inline src/components/__tests__/RouteInputTransportType.cy.js:104]: `const avatarTransportIconSize = 32`

File: src/components/__tests__/RouteInputTransportType.cy.js [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+    // icon
+    cy.dataCy('icon-transport').should('be.visible').and('have.color', primary);
```
Comment (tmszi) [inline src/components/__tests__/RouteInputTransportType.cy.js:106]: Same as my comment above.

File: src/components/__tests__/RouteItemDisplay.cy.js [hunk 8]
```diff
@@ -142,20 +156,124 @@ function coreTests() {
...
+        cy.wrap(getTransportLabel(route.transport)).should(
+          'eq',
+          description.text(),
+        );
+      });
+    });
+  });
+
+  it('renders two sections stacked', () => {
+    cy.testElementPercentageWidth(cy.dataCy('section-direction'), 100);
```
Comment (tmszi) [inline src/components/__tests__/RouteItemDisplay.cy.js:209]:
```
const sectionDirection = 'section-direction'
const sectionDirectionPercentageWidth = 100
```

File: src/components/__tests__/RouteItemDisplay.cy.js [hunk 9]
```diff
@@ -142,20 +156,124 @@ function coreTests() {
...
+      } else {
+        // distance value empty
+        cy.dataCy('label-distance').should('not.exist');
+      }
+    });
+  });
+
+  it('renders icons correctly', () => {
+    // icon direction
+    cy.dataCy('label-direction-icon', { timeout: 4000 })
```
Comment (tmszi) [inline src/components/__tests__/RouteItemDisplay.cy.js:233]: Really we need here timeout?

File: src/components/__tests__/RouteInputTransportType.cy.js [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+      .should('be.visible')
+      .and('have.css', 'font-size', '14px')
+      .and('have.css', 'font-weight', '400')
+      .and('have.color', grey10)
+      .and('contain', i18n.global.t('routes.transport.bike'));
+  });
+
+  it('renders icons correctly', () => {
+    // icons transport
+    cy.dataCy('icon-transport', { timeout: 4000 }).each((element, index) => {
```
Comment (tmszi) [inline src/components/__tests__/RouteInputTransportType.cy.js:120]: Really we need here timeout?

File: src/components/__tests__/RouteItemDisplay.cy.js [hunk 11]
```diff
@@ -142,20 +156,124 @@ function coreTests() {
...
     cy.dataCy('icon-transport')
       .should('be.visible')
-      .invoke('height')
-      .should('be.eq', 24);
-    cy.dataCy('icon-transport')
-      .should('be.visible')
-      .invoke('width')
-      .should('be.eq', 24);
+      .and('have.css', 'font-size', '18px');
+    cy.dataCy('icon-transport').invoke('height').should('be.eq', 18);
```
Comment (tmszi) [inline src/components/__tests__/RouteItemDisplay.cy.js:167]: Repeatable string or number should be const.

File: src/components/__tests__/RouteItemDisplay.cy.js [hunk 12]
```diff
@@ -142,20 +156,124 @@ function coreTests() {
       .should('be.visible')
       .invoke('width')
       .should('be.eq', 18);
+    // avatar transport
+    cy.dataCy('avatar-transport').should('be.visible');
```
Comment (tmszi) [inline src/components/__tests__/RouteItemDisplay.cy.js:160]: Repeatable string or number should be const.

File: src/components/__tests__/RouteItemDisplay.cy.js [hunk 13]
```diff
@@ -142,20 +156,124 @@ function coreTests() {
...
-      .and('have.color', black);
+      .and('have.color', grey10);
+  });
+
+  it('renders correct transport value', () => {
+    cy.get('@route').then((route) => {
+      // test styles if transport value is logged
+      if (route.transport) {
+        // transport value
+        cy.dataCy('avatar-transport')
```
Comment (tmszi) [inline src/components/__tests__/RouteItemDisplay.cy.js:182]: Repeatable string or number should be const.

File: src/components/__tests__/RouteItemDisplay.cy.js [hunk 14]
```diff
@@ -142,20 +156,124 @@ function coreTests() {
...
+      .should('be.visible')
+      .matchImageSnapshot(`${Cypress.currentTest.titlePath}-direction`, {
+        failureThreshold: 0.1,
+        failureThresholdType: 'percent',
+        timeout: 4000,
+        customDiffConfig: { threshold: 0.4 },
+        retries: 2,
+      });
+    // icon transport
+    cy.dataCy('icon-transport', { timeout: 4000 })
```
Comment (tmszi) [inline src/components/__tests__/RouteItemDisplay.cy.js:243]: Really we need timeout here?

File: src/components/__tests__/RouteListEdit.cy.js [hunk 15]
```diff
@@ -23,6 +20,10 @@ describe('<RouteListEdit>', () => {
     });
 
     coreTests();
+
+    it('renders items in two columns', () => {
+      cy.testElementPercentageWidth(cy.dataCy('route-list-item-wrapper'), 50);
```
Comment (tmszi) [inline src/components/__tests__/RouteListEdit.cy.js:25]: Repeatable string or number should be const.

File: src/components/routes/RouteInputTransportType.vue [hunk 16]
```diff
@@ -0,0 +1,160 @@
+<script lang="ts">
+/**
+ * RouteInputTransportType Component
+ *
+ * @description * Use this component to render toggle buttons for selecting
+ * transport type.
+ * This component is used in `RouteItemEdit` and `RouteCalendarPanel`.
+ *
+ * @props
+ * - `modelValue` (string, required): The object representing selected value.
```
Comment (tmszi) [inline src/components/routes/RouteInputTransportType.vue:10]: Missing prop doc.

File: src/i18n/sk.toml [hunk 17]
```diff
@@ -561,3 +561,4 @@ car = "Autom"
 none = "Nič"
```
Comment (tmszi) [inline src/i18n/sk.toml:561]: Žiadna

**Comment by tmszi** [comment]
Custom icons path should be ./public/icons/routes_calendar/icons.svg

File: src/components/__tests__/RouteInputDistance.cy.js (hunk 1 above)
Comment (maceksimon) [reply src/components/__tests__/RouteInputDistance.cy.js:80]: Added as a command.

File: src/components/__tests__/RouteInputDistance.cy.js (hunk 2 above)
Comment (maceksimon) [reply src/components/__tests__/RouteInputDistance.cy.js:159]: Extracted to a variable.

File: src/components/__tests__/RouteInputDistance.cy.js (hunk 3 above)
Comment (maceksimon) [reply src/components/__tests__/RouteInputDistance.cy.js:160]: Extracted to a variable.

File: src/components/__tests__/RouteInputTransportType.cy.js (hunk 4 above)
Comment (maceksimon) [reply src/components/__tests__/RouteInputTransportType.cy.js:56]: Replaced with a test command.

File: src/components/__tests__/RouteInputTransportType.cy.js (hunk 5 above)
Comment (maceksimon) [reply src/components/__tests__/RouteInputTransportType.cy.js:102]: Extracted to a variable.

File: src/components/__tests__/RouteInputTransportType.cy.js (hunk 6 above)
Comment (maceksimon) [reply src/components/__tests__/RouteInputTransportType.cy.js:104]: Extracted to a variable.

File: src/components/__tests__/RouteInputTransportType.cy.js (hunk 7 above)
Comment (maceksimon) [reply src/components/__tests__/RouteInputTransportType.cy.js:106]: Extracted to a variable.

File: src/components/__tests__/RouteItemDisplay.cy.js (hunk 8 above)
Comment (maceksimon) [reply src/components/__tests__/RouteItemDisplay.cy.js:209]: Extracted values and selectors into variables.

File: src/components/__tests__/RouteItemDisplay.cy.js (hunk 9 above)
Comment (maceksimon) [reply src/components/__tests__/RouteItemDisplay.cy.js:233]: This was more of a precaution. Local tests pass without it. Removed.

File: src/components/__tests__/RouteInputTransportType.cy.js (hunk 10 above)
Comment (maceksimon) [reply src/components/__tests__/RouteInputTransportType.cy.js:120]: This was more of a precaution. Local tests pass without it. Removed.

File: src/components/__tests__/RouteItemDisplay.cy.js (hunk 11 above)
Comment (maceksimon) [reply src/components/__tests__/RouteItemDisplay.cy.js:167]: Extracted to a variable.

File: src/components/__tests__/RouteItemDisplay.cy.js (hunk 12 above)
Comment (maceksimon) [reply src/components/__tests__/RouteItemDisplay.cy.js:160]: Extracted to a variable.

File: src/components/__tests__/RouteItemDisplay.cy.js (hunk 13 above)
Comment (maceksimon) [reply src/components/__tests__/RouteItemDisplay.cy.js:182]: Extracted to a variable.

File: src/components/__tests__/RouteItemDisplay.cy.js (hunk 14 above)
Comment (maceksimon) [reply src/components/__tests__/RouteItemDisplay.cy.js:243]: This was more of a precaution. Local tests pass without it. Removed.

File: src/components/__tests__/RouteListEdit.cy.js (hunk 15 above)
Comment (maceksimon) [reply src/components/__tests__/RouteListEdit.cy.js:25]: Extracted to a variable.

File: src/components/routes/RouteInputTransportType.vue (hunk 16 above)
Comment (maceksimon) [reply src/components/routes/RouteInputTransportType.vue:10]: Added.

File: src/i18n/sk.toml (hunk 17 above)
Comment (maceksimon) [reply src/i18n/sk.toml:561]: Corrected.

**Comment by maceksimon** [comment]
SVG icons are moved to the correct folder.

File: src/components/__tests__/RouteItemDisplay.cy.js [hunk 18]
```diff
@@ -15,147 +43,272 @@ describe('<RouteItemDisplay>', () => {
...
+      }
+    });
+  });
+
+  it('renders icons correctly', () => {
+    // icon direction
+    cy.dataCy(selectorDirectionIcon)
+      .should('be.visible')
+      .matchImageSnapshot(`${Cypress.currentTest.titlePath}-direction`, {
+        failureThreshold: 0.1,
```
Comment (tmszi) [inline src/components/__tests__/RouteItemDisplay.cy.js:271]: params args repeat across tests

Comment (maceksimon) [reply src/components/__tests__/RouteItemDisplay.cy.js]: Extracted function parameter to a variable.

**Comment by tmszi** [comment]
Pointer with "do not select" icon, is allowing select route in the calendar.

route_selection_on_deactivated_route

**Comment by maceksimon** [comment]
> Pointer with "do not select" icon, is allowing select route in the calendar.

Fixed. Items outside current month are disabled via a prop in `CalendarItemDisplay`.

**Comment by maceksimon** [comment]
> It is possible typing negative number or zero into Input distance input widget (By bike, On foot, By public transport). Should not be here a input value validator?

Updated. Added a conditional validation (enabled by default). This will most likely be used in [list section](https://www.figma.com/design/L8dVREySVXxh3X12TcFDdR/Do-pr%C3%A1ce-na-kole?node-id=4858-104042&t=30orbYgabbcBdpGQ-1) where user can edit missing routes (we might want to allow logging only some of them).

**Comment by tmszi** [comment]
> > It is possible typing negative number or zero into Input distance input widget (By bike, On foot, By public transport). Should not be here a input value validator?
> 
> Updated. Added a conditional validation (enabled by default). This will most likely be used in [list section](https://www.figma.com/design/L8dVREySVXxh3X12TcFDdR/Do-pr%C3%A1ce-na-kole?node-id=4858-104042&t=30orbYgabbcBdpGQ-1) where user can edit missing routes (we might want to allow logging only some of them).

I tried again, but input widget allow me typing negative number. Is it really working as expected?

**Comment by maceksimon** [comment]
> I tried again, but input widget allow me typing negative number. Is it really working as expected?

My mistake, negative numbers were not validated. Added validation + test.

**Pull request merged**